src/java.desktop/share/classes/javax/swing/plaf/synth/SynthParser.java

```diff
@@ -255,17 +246,7 @@ public void parse(InputStream inputStream,
      * Returns the path to a resource.
      */
     private URL getResource(String path) {
-        if (_classResourceBase != null) {
-            return _classResourceBase.getResource(path);
-        } else {
-            try {
-                @SuppressWarnings("deprecation")
-                var result = new URL(_urlResourceBase, path);
-                return result;
-            } catch (MalformedURLException mue) {
-                return null;
-            }
-        }
+        return _classResourceBase.getResource(path);
```

Comment:
Any particular reason for removing the null check?

Reply:
Yes. the resource was *either* url-based or classpath based. One would be null, the other not.
Now there's only one, so it must be the non-null one.